src/main/scala/com/intenthq/sbt/ThriftPlugin.scala

```diff
@@ -58,77 +59,92 @@ object ThriftPlugin extends AutoPlugin {
   import autoImport._
 
   override def requires = sbt.plugins.JvmPlugin
-
   override def trigger = allRequirements
 
   val Thrift = config("thrift")
 
   val thriftSettings: Seq[Setting[_]] = inConfig(Thrift)(Seq(
     thrift := "thrift",
-    thriftSourceDir <<= sourceDirectory { _ / "main" / "thrift" },
+    thriftSourceDir := { sourceDirectory.value / "main" / "thrift" },
     thriftJavaEnabled := true,
     thriftJavaOptions := Seq(),
-    thriftOutputDir <<= sourceManaged { _ / "main" },
-    thriftGenerateJava <<= (thriftJavaEnabled, thriftSourceDir, thriftOutputDir, thrift, thriftJavaOptions, streams).map { (te, tsd, tod, t, to, s) =>
-      if (te) {
-        compileThrift(tsd, tod, t, "java", to, s.log, s.cacheDirectory / "thrift-java") }
+    thriftOutputDir := { sourceManaged.value / "main" },
+    thriftGenerateJava := (Def.taskDyn {
+      if (thriftJavaEnabled.value)
+        Def.task {
+          compileThrift(thriftSourceDir.value, thriftOutputDir.value, thrift.value, "java", thriftJavaOptions.value, streams.value.log, streams.value.cacheDirectory / "thrift-java")
+        }
       else
-        Seq()
-    },
+        Def.task {
+          Seq.empty[File]
+        }
+    }).value,
     thriftJsEnabled := false,
     thriftJsOptions := Seq(),
     thriftJsOutputDir := file("target/gen-js"),
-    thriftGenerateJs <<= (thriftJsEnabled, thriftSourceDir, thriftJsOutputDir, thrift, thriftJsOptions, streams).map { (te, tsd, tod, t, to, s) =>
-      if (te) {
-        compileThrift(tsd, tod, t, "js", to, s.log, s.cacheDirectory / "thrift-js") }
-        Seq()
-    },
+    thriftGenerateJs := (Def.taskDyn {
```

Comment:
If a task is conditional on its action, it must be wrapped in a dynamic task. This is a new thing and it's explained here: http://www.scala-sbt.org/1.x/docs/Howto-Dynamic-Task.html